### PR TITLE
Removed temporary <1.29 bug workaround for nil affinity

### PR DESCRIPTION
**Pull request by AnnaZivkovic**
[MIXEDARCH-430](https://issues.redhat.com//browse/MIXEDARCH-430): Removed temporary <1.29 bug workaround for nil affinity

### CODE REVIEW

**Comment by aleskandro** [comment]
Hi @AnnaZivkovic, we'd also remove the call to `WithAffinity()` in https://github.com/openshift/multiarch-manager-operator/blob/f2315e12e4a1a10f7b71f1b3dd16277b55813aad/controllers/podplacement/pod_model_test.go#L544-L545

File: controllers/podplacement/pod_model_test.go [hunk 1]
```diff
@@ -540,18 +540,6 @@ func TestPod_SetNodeAffinityArchRequirement(t *testing.T) {
 					},
 				}).Build(),
 		},
-		{
-			name: "pod with no node affinity",
-			pod:  NewPod().WithContainersImages(fake.MultiArchImage).WithAffinity(nil).Build(),
```
Comment (aleskandro) [inline controllers/podplacement/pod_model_test.go:545]: Hey @AnnaZivkovic, I wouldn't remove this test, but just the WithAffinity call.

```
NewPod().WithContainersImages(fake.MultiArchImage).Build()
```

**Comment by aleskandro** [comment]
@AnnaZivkovic maybe you need to rebase due to changes in https://github.com/openshift/multiarch-tuning-operator/pull/48

**Comment by Prashanth684** [comment]
This  [TODO](https://github.com/openshift/multiarch-tuning-operator/blob/main/pkg/e2e/podplacement/pod_placement_test.go#L240C40-L240C67) needs to be addressed for the test to pass

**Comment by aleskandro** [comment]
/test ci/prow/e2e-gcp-multi-operator

(just to check an issue with the other PR #50)

**Pull request merged**